src/libraries/Microsoft.Extensions.Hosting/src/HostApplicationBuilder.cs

```diff
@@ -88,7 +88,12 @@ public HostApplicationBuilder(HostApplicationBuilderSettings? settings)
 
             if (!settings.DisableDefaults)
             {
-                HostingHostBuilderExtensions.ApplyDefaultHostConfiguration(Configuration, settings.Args);
+                if (Configuration[HostDefaults.ContentRootKey] is null)
```

Comment:
This check helps because ASPNETCORE_CONTENTROOT is already set before creating the HostApplicationBuilder?

ASPNETCORE_CONTENTROOT would still get overridden by DONET_CONTENTROOT, right?

Reply:
It would also still get overriden by `settings.ContentRootPath` on line 114 below, right?

Reply:
Correct. `ASPNETCORE_CONTENTROOT` still gets overridden by `DOTNET_CONTENTROOT` and `settings.ContentRootPath`, but it no longer gets overridden by the CWD.

I almost wrote the check like this instead.

```suggestion
                if (settings.ContentRootPath is null && Configuration[HostDefaults.ContentRootKey] is null)
```

I think I'll just apply this suggestion. It doesn't change behavior, but it saves a negligible amount of work in this case.